internal/services/compute/orchestrated_virtual_machine_scale_set_resource.go

```diff
@@ -126,29 +252,578 @@ func resourceOrchestratedVirtualMachineScaleSetCreateUpdate(d *pluginsdk.Resourc
+		}
+
+		virtualMachineProfile.BillingProfile = &compute.BillingProfile{
+			MaxPrice: utils.Float(v),
+		}
+	}
+
+	if v, ok := d.GetOk("encryption_at_host_enabled"); ok {
+		virtualMachineProfile.SecurityProfile = &compute.SecurityProfile{
+			EncryptionAtHost: utils.Bool(v.(bool)),
+		}
+	}
+
+	if v, ok := d.GetOk("eviction_policy"); ok {
+		if virtualMachineProfile.Priority != compute.VirtualMachinePriorityTypesSpot {
+			return fmt.Errorf("an `eviction_policy` can only be specified when `priority` is set to `Spot`")
+		}
+		virtualMachineProfile.EvictionPolicy = compute.VirtualMachineEvictionPolicyTypes(v.(string))
+	} else if virtualMachineProfile.Priority == compute.VirtualMachinePriorityTypesSpot {
+		return fmt.Errorf("an `eviction_policy` must be specified when `priority` is set to `Spot`")
+	}
+
+	if v, ok := d.GetOk("license_type"); ok {
+		virtualMachineProfile.LicenseType = utils.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("termination_notification"); ok {
+		virtualMachineProfile.ScheduledEventsProfile = ExpandVirtualMachineScaleSetScheduledEventsProfile(v.([]interface{}))
+	}
+
+	// Only inclued the virtual machine profile if this is not a legacy configuration
+	if !isLegacy {
+		if v, ok := d.GetOk("plan"); ok {
+			props.Plan = expandPlan(v.([]interface{}))
+		}
+
+		if v, ok := d.GetOk("identity"); ok {
+			identity, err := ExpandVirtualMachineScaleSetIdentity(v.([]interface{}))
+			if err != nil {
+				return fmt.Errorf("expanding `identity`: %+v", err)
+			}
+			props.Identity = identity
+		}
+
+		if v, ok := d.GetOk("automatic_instance_repair"); ok {
+			props.VirtualMachineScaleSetProperties.AutomaticRepairsPolicy = ExpandOrchestratedVirtualMachineScaleSetAutomaticRepairsPolicy(v.([]interface{}))
+		}
+
+		if v, ok := d.GetOk("zone_balance"); ok && v.(bool) {
+			if len(*zones) == 0 {
+				return fmt.Errorf("`zone_balance` can only be set to `true` when zones are specified")
+			}
+
+			props.VirtualMachineScaleSetProperties.ZoneBalance = utils.Bool(v.(bool))
+		}
+
+		props.VirtualMachineScaleSetProperties.VirtualMachineProfile = &virtualMachineProfile
+	}
+
+	log.Printf("[DEBUG] Creating Orchestrated Virtual Machine Scale Set %q (Resource Group %q)..", name, resourceGroup)
 	future, err := client.CreateOrUpdate(ctx, resourceGroup, name, props)
 	if err != nil {
 		return fmt.Errorf("creating Orchestrated Virtual Machine Scale Set %q (Resource Group %q): %+v", name, resourceGroup, err)
 	}
 
+	log.Printf("[DEBUG] Waiting for Orchestrated Virtual Machine Scale Set %q (Resource Group %q) to be created..", name, resourceGroup)
 	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
-		return fmt.Errorf("waiting for creation of Orchestrated Virtual Machine Scale Set %q (Resource Group %q): %+v", name, resourceGroup, err)
+		// If it is a Retryable Error re-issue the PUT for 10 loops until either the error goes away or the limit has been reached
+		if strings.Contains(err.Error(), "RetryableError") {
```

Comment:
the Azure SDK for Go (go-autorest) should be doing this automatically - so we can remove this and file a bug in the Azure SDK to get that fixed if it doesn't?

Reply:
I totally agree with you, they should, but they are not so I added this as a workaround because if I don't you end up with a resource that cannot be deleted because the subnet is in use. I will follow up with the SDK team about this issue.